src/workerd/jsg/jsvalue.c++

```diff
@@ -536,16 +536,11 @@ JsObject Lock::obj() {
 
 JsObject Lock::obj(kj::ArrayPtr<kj::StringPtr> keys, kj::ArrayPtr<JsValue> values) {
   KJ_DASSERT(keys.size() == values.size());
-  v8::LocalVector<v8::Name> keys_(v8Isolate);
-  v8::LocalVector<v8::Value> values_(v8Isolate);
-  keys_.reserve(keys.size());
-  values_.reserve(keys.size());
-  for (auto& k: keys) {
-    v8::Local<v8::String> key = str(k);
-    keys_.push_back(key);
-  }
-  for (auto& v: values) {
-    values_.push_back(v);
+  v8::LocalVector<v8::Name> keys_(v8Isolate, keys.size());
+  v8::LocalVector<v8::Value> values_(v8Isolate, keys.size());
+  for (size_t i = 0; i < keys.size(); i++) {
+    keys_[i] = strIntern(keys[i]).inner;
+    values_[i] = values[i];
```

Comment:
The fact that this variant returns an object with a null prototype might be breaking in some cases